hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockNetworkingService.java

```diff
@@ -278,6 +278,14 @@ public void onConnectionClose(MockConnection connection, Throwable t, boolean si
                 if (!ns.mapConnections.remove(endPoint, connection)) {
                     return;
                 }
+
+                NetworkingService remoteNetworkingService = connection.remoteNodeEngine.getNode().getNetworkingService();
+                Connection remoteConnection = remoteNetworkingService.getEndpointManager(MEMBER)
```

Comment:
Is this going to work with the advanced network configuration as well if there is a WAN endpoint defined? Shouldn't something like `EndpointQualifier.resolve(WAN, "wanCfg")` be used in that case? Looks like the `WanConnectionManager` in the EE side registers the listener with WAN endpoint qualifier if the endpoint is named: https://github.com/hazelcast/hazelcast-enterprise/pull/2792/files#diff-bb60c7b6f88817df95a2f1bbf288162dR102

Reply:
I saw that the mock implementation completely ignores the qualifier which is why I haven't given it much thought. See [here](https://github.com/hazelcast/hazelcast/blob/92ca85687ad1f9cc2057a7cd3b7a27f22bad8019/hazelcast/src/test/java/com/hazelcast/nio/tcp/FirewallingNetworkingService.java#L85-L94) and [here](https://github.com/hazelcast/hazelcast/blob/1d2c6c49858a84199ef8eb8a3ded6c7fe1c196b0/hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockNetworkingService.java#L328-L331). If you have any further questions, I guess @vbekiaris or @tkountis might help :)

Reply:
Ok, that's fair :) Maybe we can add a comment to clarify that it's not taken into account, or just introduce an `IGNORED_QUALIFIER`.

Reply:
I'll add a comment and change the value to `null`, to further signal that it's irrelevant. Once the mock connection manager is changed to use the qualifier, we should be notified by NPE's being thrown from WAN tests.